src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5874,6 +5883,66 @@ private ScanTypeFlags ScanTupleType(out SyntaxToken lastTokenOfType)
             return ScanTypeFlags.NotType;
         }
 
+#nullable enable
+        private ScanTypeFlags ScanFunctionPointerType(out SyntaxToken lastTokenOfType)
+        {
+            Debug.Assert(IsFunctionPointerStart());
+            _ = EatContextualToken(SyntaxKind.DelegateKeyword);
+            lastTokenOfType = EatToken(SyntaxKind.AsteriskToken);
+
+            if (IsPossibleCallingConvention())
+            {
+                _ = EatToken();
```

Comment:
>_ = EatToken(); [](start = 16, length = 15)

I think the code would be more robust if we adjust lastTokenOfType as we go along, rather than making an assumption that, if we get here, we will not get into the ```if``` below and return early because **the current** implementation of ```IsPossibleCallingConvention``` ensures that ```IsPossibleFunctionPointerParameterListStart(CurrentToken)``` is true after this line.